### DISCUSSION TITLE
Paginate

### DISCUSSION THREAD

**Discussion by OmarISE** [livewire/livewire#8234] [1. Help]
Property type not supported in Livewire for property
![Screenshot (633)](https://github.com/livewire/livewire/assets/153657027/8641c543-4a3c-4a82-93d1-d7abfc4188e0)

**it is all my codes, there is no problem with my codes just with paginate**

 use WithPagination;
    public $properties;
    public $type = 'all';
    public $purpose;
    public $max;
    public $min;
    public $city_id;
    public $street_id;

    public function mount()
    {
        $this->search();
    }

    public function search()
    {
        $query = Property::query()->where('status', 1);

        if ($this->type !== 'all') {
            $query->where('type', $this->type);
        }

        if ($this->city_id) {
            $query->whereHas('city', function ($query) {
                $query->where('id', 'like', '%' . $this->city_id . '%');
            });
        }

        if ($this->street_id) {
            $query->whereHas('street', function ($query) {
                $query->where('id', 'like', '%' . $this->street_id . '%');
            });
        }
        if ($this->purpose) {
            $query->where('purpose', 'like', '%' . $this->purpose . '%');
        }

        if ($this->min && $this->max) {
            $query->whereBetween('price', [$this->min, $this->max]);
        } elseif ($this->min) {
            $query->where('price', '>=', $this->min);
        } elseif ($this->max) {
            $query->where('price', '<=', $this->max);
        }

        $this->properties = $query->paginate(5);
    }
    public function setType($type)
    {
        $this->type = $type;
        $this->search();
    }

    #[Computed()]
    public function cities()
    {
        return City::all();
    }
    #[Computed()]
    public function streets()
    {
        return Street::Street($this->city_id);
    }
    public function render()
    {

        return view('livewire.test');
    }

**Comment by joshhanley** [comment]
@OmarISE you can't assign paginated collections to public properties. Instead you should use a computed property https://livewire.laravel.com/docs/computed-properties

```php
    use WithPagination;
    public $type = 'all';
    public $purpose;
    public $max;
    public $min;
    public $city_id;
    public $street_id;

    #[Computed]
    public function properties()
    {
        $query = Property::query()->where('status', 1);

        if ($this->type !== 'all') {
            $query->where('type', $this->type);
        }

        if ($this->city_id) {
            $query->whereHas('city', function ($query) {
                $query->where('id', 'like', '%' . $this->city_id . '%');
            });
        }

        if ($this->street_id) {
            $query->whereHas('street', function ($query) {
                $query->where('id', 'like', '%' . $this->street_id . '%');
            });
        }
        if ($this->purpose) {
            $query->where('purpose', 'like', '%' . $this->purpose . '%');
        }

        if ($this->min && $this->max) {
            $query->whereBetween('price', [$this->min, $this->max]);
        } elseif ($this->min) {
            $query->where('price', '>=', $this->min);
        } elseif ($this->max) {
            $query->where('price', '<=', $this->max);
        }

        return $query->paginate(5);
    }

    public function setType($type)
    {
        $this->type = $type;
        $this->search();
    }

    #[Computed()]
    public function cities()
    {
        return City::all();
    }

    #[Computed()]
    public function streets()
    {
        return Street::Street($this->city_id);
    }

    public function render()
    {
        return view('livewire.test');
    }
```

Then make sure you change your blade view from `$properties` to `$this->properties`.

**Reply by OmarISE** [reply]
Thank you so much for your help. I really appreciate it
but i have a form for searching i fixed like this


 use WithPagination;
    public $type = 'all';
    public $purpose;
    public $max;
    public $min;
    public $city_id;
    public $street_id;


    public function mount()
    {
        $this->search();
    }


    public function search()
    {
        $this->applaySearch();
        $this->resetPage();
    }

    #[Computed()]
    public function applaySearch()
    {
        $query = Property::query()->Status();

        if ($this->type !== 'all') {
            $query->where('type', $this->type);
        }

        if ($this->city_id !== '') {
            $query->whereHas('city', function ($query) {
                $query->where('id', 'like', '%' . $this->city_id . '%');
            });
        }

        if ($this->street_id !== '') {
            $query->whereHas('street', function ($query) {
                $query->where('id', 'like', '%' . $this->street_id . '%');
            });
        }

        if ($this->purpose !== '') {
            $query->where('purpose', 'like', '%' . $this->purpose . '%');
        }

        if ($this->min && $this->max) {
            $query->whereBetween('price', [$this->min, $this->max]);
        } elseif ($this->min) {
            $query->where('price', '>=', $this->min);
        } elseif ($this->max) {
            $query->where('price', '<=', $this->max);
        }
        return $query;
        $this->reset('city_id');
    }


    public function updatedsearch()
    {
        $this->resetPage();
    }


    #[Computed(key: 'main-page-index')]
    public function properties()
    {
        $o = $this->applaySearch()->latest();
        return $o->paginate(3);
    }


    public function setType($type)
    {
        $this->type = $type;
        $this->search();
    }



    #[Computed(key: 'main-page-index')]
    public function cities()
    {
        return City::all();
    }
    #[Computed(key: 'main-page-index')]
    public function streets()
    {
        return Street::Street($this->city_id);
    }

**Reply by joshhanley** [reply]
@OmarISE no worries! 🙂

You shouldn't call `$this->applaySearch()` manually as it is a computed property. It will be called for you when you access the computed property. So you can remove the `mount()` method all together.

Also calling `$o = $this->applaySearch()->latest(); return $o->paginate(3);` isn't using database pagination, so you should get rid of the `latest()` call and instead call paginate directly.

Personally I would move everything in `applaySearch()` back into the `properties` computed property and leave it like that.
